File: server/languages/php/engine.php
<?php
// Class Engine
// Read and write to/form server proxy
class Engine
{
    public static function send($commandJS){
        
        fwrite(STDERR, "none\n");
        fflush(STDERR);

        fwrite(STDOUT,"start\n");
        fflush(STDOUT);

        fwrite(STDOUT, $commandJS."\n");
        fflush(STDOUT);

        fwrite(STDOUT,"insert\n");
        fflush(STDOUT);

        $stdin = fgets(STDIN);
        fflush(STDIN);

        fwrite(STDOUT,"close\n");
        fflush(STDOUT);

        return str_replace("\n","", $stdin);
    }
}

?>

File: server/languages/ruby/engine.rb
# Class Engine
# read and write to/form server proxy

class Engine
    def self.send(commandJS)
        
        STDERR.puts "none\n"
        STDERR.flush
        
        puts "start"
        STDOUT.flush

        puts commandJS
        STDOUT.flush

        puts "insert"
        STDOUT.flush

        msg = STDIN.gets
        STDIN.flush

        puts "close"
        STDOUT.flush

        msg.gsub("\n","")
    end
end